### PR TITLE
Add a refresh button to the serverlist

### PR DESCRIPTION
**Pull request by ThomasMonroe314**
closes #6048 and #6028

### CODE REVIEW

**Comment by SmallJoker** [comment]
> Unicode is fine too, it should just not take lots of space (as it does now).

https://github.com/minetest/minetest/pull/6048#issuecomment-353055872

Instead of `1.5` "units", this PR now changes it to `2.0`. As I did the math, it's now even 33.3% bigger - too much IMO.

**Comment by ThomasMonroe314** [comment]
@SmallJoker I saw that and was working on making them smaller

**Comment by rubenwardy** [comment]
This is exactly the same as the other PR

**Comment by ThomasMonroe314** [comment]
Images don't work, me and @Ezhh have tried over and over.
I'm not sure if its a bug or not, but as @paramat said text is more consistent.

**Comment by Ezhh** [comment]
If the imagebuttons genuinely don't currently work on the main menu (it's still possible my testing was wrong), then the real fix would probably be getting them to work.

**Comment by Megaf** [comment]
:+1:

**Comment by jastevenson303** [comment]
@megaf that's what they added Reactions for.  So if all you wanted to add
was a thumbsup, that you could click the thumbsup reaction on OP.

**Comment by paramat** [comment]
Text is ok for me and clearer in meaning.

**Comment by Ezhh** [comment]
If going with text, I think the buttons here might need to be a tiny bit closer together (by enlarging the search button or box, not the refresh button). The horizontal spacing looks too much compared to the rest of the screen.

**Comment by paramat** [comment]
^ Yes, for the spaces between search box, 'search' and 'reset' use the spacing found between 'name' and 'password'.
Resize the screen radically (from 800x600 up) to check the text still fits in the boxes.

**Comment by ThomasMonroe314** [comment]
> The horizontal spacing looks too much compared to the rest of the screen.

@Ezhh, Yeah I thought it looked a little weird.

> Resize the screen radically (from 800x600 up) to check the text still fits in the boxes.

Will do

**Comment by Ezhh** [comment]
When I worked on this myself, I had the search box and Search button, and Search button and Refresh button have approximately the same amount of space between them as is between the Address and Port field.

**Comment by ThomasMonroe314** [comment]
Ok, I just adjusted them to have about as much space as between the the Del. Favorite and Connect button if that's fine with you.

screenshot-3

**Comment by Ezhh** [comment]
Looks better to me, but it still comes down to whether text button will be accepted.

**Comment by ThomasMonroe314** [comment]
Ok good, text is also more consistent as paramat said.

**Comment by paramat** [comment]
That spacing looks good to me.

**Comment by ThomasMonroe314** [comment]
Personally I like both the image and the text, so I'm good either way.

screenshot-4

**Comment by paramat** [comment]
See https://github.com/minetest/minetest_game/pull/2020
You need to blur the edges of the texture slightly to reduce edge distortion when resizing the game window.

**Comment by rubenwardy** [comment]
You should also decrease the brightness from white to #f0f0f0 imo

**Comment by ThomasMonroe314** [comment]
Okay I'll do that real quick.

**Comment by paramat** [comment]
In Gimp: Filters > Blur > Blur

**Comment by rubenwardy** [comment]
Where did you find this icon from? You need attribution and license

I'd also prefer one which is more standard, such as the material refresh icon (Apache): https://material.io/icons/#ic_refresh

**Comment by paramat** [comment]
An image button looks fine to me, and saves space.

**Comment by ThomasMonroe314** [comment]
icon updated and blurred to the right amount.

screenshot

**Comment by Ezhh** [comment]
It might just be the screenshot... but it's looking very uneven?

**Comment by ThomasMonroe314** [comment]
yeah it kinda looks like an octagon, idk why though. Might have something to do with scaling. I'll play around with it.

**Comment by paramat** [comment]
It's distorting because although it's blurred it is now only 48px, it should be 128px and blurred. You need to start from a 128px original.

**Comment by ThomasMonroe314** [comment]
Ok I'll see if I can find a 128px version.

**Comment by rubenwardy** [comment]
I prefer the current one (but use a higher res) as it's more standard

> Ok I'll see if I can find a 128px version.

Simply increase the DP on the download, or download and render the SVG

**Comment by paramat** [comment]
Yes a 128px standard style one is fine for me.

**Comment by ThomasMonroe314** [comment]
icon updated at 128px

screenshot-7

**Comment by paramat** [comment]
Seems good.

**Comment by Ezhh** [comment]
Looks much better. The only thing I might still change is how much of a border is around the image on the button (I'd try to align with how much space is between the top/bottom of the text and the button edge - but I admit I'm fussy about these things).

**Comment by ThomasMonroe314** [comment]
@Ezhh so scale the arrow down slightly?

**Comment by Ezhh** [comment]
I think it would look better.

**Comment by ThomasMonroe314** [comment]
so the button needs to be bigger?

**Comment by Ezhh** [comment]
No. The amount of button that is image would need to be less. I assume adding padding to the image would achieve this, but haven't tested myself.

**Comment by red-001** [comment]
seems a bit out of place, with the search button, maybe edit the search button text to be in bold.

**Comment by ThomasMonroe314** [comment]
idk how to get 'padding' without scaling the arrow down some

@red-001 then I would need to do that to every button on the main menu.

**Comment by red-001** [comment]
then change the image to get it to fit in a bit better?

**Comment by paramat** [comment]
As in, enlarge the texture not by scaling but by adding a border around it.
In Gimp:
image > canvas size (use 'centre' feature)
then
layer > layer to image size

**Comment by Ezhh** [comment]
It'll look less out of place by adding the padding/border. There's no need to bold anything.

**Comment by paramat** [comment]
@ThomasMonroe314 if you can't figure out adding padding i can do this quickly.

**Comment by ThomasMonroe314** [comment]
@paramat ok, you can probably do it better than I can too

**Comment by paramat** [comment]
refresh136

^ 136px

refresh144

^ 144px

refresh152

^ 152px

Could you show a screenshot for each?

**Comment by ThomasMonroe314** [comment]
136px:
screenshot-3
144px:
screenshot-2
152px:
screenshot-1

**Comment by paramat** [comment]
Thanks. I agree 152px looks best and much better than 128px.
@rubenwardy is that ok with you?

**Comment by Megaf** [comment]
Am I the only one who prefers a text button?
Besides, we don't have any button with image in the menu, it's all text, another reason I believe it should be text based.

**Comment by rubenwardy** [comment]
Lgtm, you just need to add the license to LICENSE.txt

File: builtin/mainmenu/tab_online.lua [hunk 1]
```diff
@@ -33,8 +33,10 @@ local function get_formspec(tabview, name, tabdata)
 
 	local retval =
 		-- Search
-		"field[0.15,0.075;6.05,1;te_search;;"..core.formspec_escape(tabdata.search_for).."]"..
-		"button[5.8,-0.25;2,1;btn_mp_search;" .. fgettext("Search") .. "]" ..
+		"field[0.15,0.075;5.91,1;te_search;;"..core.formspec_escape(tabdata.search_for).."]"..
+		"button[5.62,-0.25;1.5,1;btn_mp_search;" .. fgettext("Search") .. "]" ..
+		"image_button[6.97,-.165;.83,.83;" .. core.formspec_escape(defaulttexturedir .. "refresh.png") 
+			.. ";btn_mp_refresh;".. fgettext("") .."]" ..
```
Comment (paramat) [inline builtin/mainmenu/tab_online.lua]: For the 4 lines above always have a space either side of '..'

Comment (paramat) [reply builtin/mainmenu/tab_online.lua]: Done by SmallJoker.

**Comment by paramat** [comment]
@sfan5 @nerzhul any approval?

**Comment by nerzhul** [comment]
I like this version

**Comment by Fixer-007** [comment]
I have a regression with it, text over this refresh button:
36508838-84c1d7a2-1766-11e8-9ed6-8014471c115e

**Comment by paramat** [comment]
'..cts/mi..' what text could that be? That will be a hint. I can't reproduce this yet.

**Comment by Fixer-007** [comment]
default
ects/mine

**Pull request merged**